src/common/winterStorageAreaCard/WinterStorageAreaCard.tsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import classNames from 'classnames';
+
+import Card from '../card/Card';
+import styles from '../harborCard/harborCard.module.scss';
+import CardHeader from '../cardHeader/CardHeader';
+import Text from '../text/Text';
+import CardBody from '../cardBody/CardBody';
+import placeholder from './winterStorageAreaPlaceholder.svg';
+import Section from '../section/Section';
+import ExternalLink from '../externalLink/ExternalLink';
+import HarborMapLinks from '../harborMapLinks/HarborMapLinks';
+import Grid from '../grid/Grid';
+import Property from '../property/Property';
+import { IconDollyEmpty, IconFence, IconPlug, IconTrestle, IconWaterTap, IconBoat } from '../icons';
+import { WinterStorageArea } from '../../features/winterStorageAreaView/types';
+import { getProductServiceTKey } from '../utils/translations';
+import { ProductServiceType } from '../../@types/__generated__/globalTypes';
+
+interface WinterStorageAreaCardProps extends WinterStorageArea {
+  className?: string;
+}
+
+const WinterStorageAreaCard = ({
+  className,
+  name,
+  servicemapId,
+  imageFile,
+  streetAddress,
+  zipCode,
+  municipality,
+  numberOfCustomers,
+  gate,
+  water,
+  electricity,
+  summerStorageForBoats,
+  summerStorageForDockingEquipment,
+  summerStorageForTrailers,
+  maps,
+}: WinterStorageAreaCardProps) => {
+  const { t } = useTranslation();
+  const serviceMapUrl = `${process.env.REACT_APP_SERVICE_MAP_URI}${servicemapId}`;
+
+  return (
+    <Card className={classNames(className, styles.card)}>
+      <CardHeader title={t('winterStorageAreaCard.title')} />
+      <CardBody>
+        <div className={styles.cardBody}>
+          <div className={styles.details}>
+            <div className={styles.imageWrapper}>
+              <img
+                alt={t('winterStorageAreaCard.imgAlt')}
+                src={imageFile ? imageFile : placeholder}
+                className={styles.image}
+              />
+            </div>
+            <div>
+              <Section>
+                <Text as="h2" size="xxl">
+                  {name}
+                </Text>
+                <Text>
+                  {streetAddress ? `${streetAddress}, ${zipCode} ${municipality}` : `${zipCode} ${municipality}`}
+                </Text>
+              </Section>
+              <Section>
+                <ExternalLink href={serviceMapUrl} variant="withArrow">
+                  {t('common.terminology.serviceMap')}
+                </ExternalLink>
+              </Section>
+              <HarborMapLinks maps={maps} />
+            </div>
+          </div>
+          <Grid colsCount={6} className={styles.propsGrid}>
```

Comment:
To make the grid columns of equal width, use this style instead:

```css
grid-template-columns: repeat(6, minmax(0, 1fr));
```

Then add these to Property label styles to make the text wrap and hyphenate:

```css
overflow-wrap: break-word;
width: 100%;
hyphens: auto;
```

Hyphenation will be super weird by default, but you can mark the intended hyphenation places with soft hyphens in the i18n files:

```
"dockingEquipmentSummerStorage": "Telakointi\u00ADtarvikkeiden kesäsäilytys"
```

Reply:
Added these improvements 👍